parser/src/lexer.rs

```diff
@@ -513,15 +512,15 @@ impl<'input, I> Lexer<'input, I>
                    -> ParseResult<Token<&'input str>, LocatedStream<I>> {
         loop {
             // Skip all whitespace before the token
-            let (_, new_input) = try!(spaces().parse_lazy(input).into());
+            let (_, new_input) = spaces().parse_lazy(input).into()?;
             input = new_input.into_inner();
```

Comment:
Looks like inference fails here because the preceding `into()` call cannot figure out that it should be turned into a `Result` :/